contracts/l2/dai.sol

```diff
@@ -72,9 +72,6 @@ contract Dai {
     assembly {chainId := chainid()}
     deploymentChainId = chainId;
     _DOMAIN_SEPARATOR = _calculateDomainSeparator(chainId);
-
-    // Set addresses which disallow transfer
-    balanceOf[address(this)] = balanceOf[address(0)] = type(uint256).max;
```

Comment:
has been reached consensus on this?

Reply:
This whole PR is removing the overflow check. So we first need to decide if this PR is okay, and if so whether we want to re-add the disallow send to contract check. I'm going to be grabbing some numbers first to aid with this decision.